app/src/main/java/ar/scacchipa/twittercloneapp/domain/CheckCredentialsUseCase.kt

```diff
@@ -0,0 +1,12 @@
+package ar.scacchipa.twittercloneapp.domain
+
+import android.content.SharedPreferences
+import ar.scacchipa.twittercloneapp.utils.Constants
+
+class CheckCredentialsUseCase(
+    private val sharedPrefs: SharedPreferences
```

Comment:
mmm I dont think this is ok. Use cases should not access a data source like a sharedPref directly. Instead there should be a repository in the middle. The use case should be composed with that repository and the repositoy will be the one in charge to talk with the data source, the sharedPref in this case.